src/uid.cpp

```diff
@@ -0,0 +1,4 @@
+#include "uid.h"
+
+template <>
+uid_factory<UID_ITEM> *uid_factory<UID_ITEM>::instance = nullptr;
```

Comment:
It's not necessary to use the factory instance for deserialization, it's not running the uid algorithm, just deserializing, so it can just be a static method. The only reason to even do that is to avoid exposing a constructor, which we might want for unit tests anyway.